scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOFlattenedInput.scala

```diff
@@ -43,11 +48,33 @@ trait H2OMOJOFlattenedInput {
     }
   }
 
+  protected def getRelevantColumnNames(flatDataFrame: DataFrame, inputs: Array[String]): Array[String] = {
+    val result = ArrayBuffer[String]()
+
+    def addIfPrefixExists(prefix: String): Unit = {
+      if (inputs.exists(i => i.startsWith(prefix + ".") && Try(i.substring(prefix.length + 1).toInt).isSuccess)) {
```

Comment:
Nit: Just thinking out loud -> should we also test edge cases such as array of array and vector of vectors?

Reply:
> vector of vectors
This is not possible. Vector is an array of doubles.

> array of array
There is already a logic in codebase that could do that, but as far i remember we didn't use it because of making predictions on streamed dataframes. To be able to deeply nested structures, we would need to materialize the whole dataframe to get array sizes.

Reply:
I see, all good then!